keps/sig-apps/2307-job-tracking-without-lingering-pods/README.md

```diff
@@ -307,21 +319,89 @@ finalizer.
 The job controller adds the finalizer in the same patch request that modifies
 the owner reference.
 
+## Monitoring Pods with finalizers
+
+Starting in 1.26, the metric `job_pod_tracking_finalizer` is a gauge that
+tracks the number of pods that currently have a job tracking finalizer.
+
+The metric increments when the job controller observes a pod created or adopted,
+and decrements when the job controller observes an update that removes the
+finalizer or a pod deletion.
+
+## Migrating Jobs with legacy tracking
+
+Starting in 1.26, when the feature gate `MigrateJobLegacyTracking` is enabled,
```

Comment:
Worth adding the reason for this which is to facilitate the migration of jobs that may run across two upgrades. It looks like an overkill since jobs are not like deployments in that they don't live forever, but there are probably cases where people upgrade their clusters back to back.

Reply:
Added notes